### PR TITLE
Fixes the Fence problems.

### LINKED ISSUE #177: Citizen stuck on fence
### Minecolonies version
develop
### Expected behavior
Citizen walks down from fence
### Actual behaviour
Citizen spins on the fence and goes nowhere
### Steps to reproduce the problem
Place fences with stairs on one side and watch citizens get trapped

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #177

# Changes proposed in this pull request:
- Fixes the fence orientation problem

I elaborated this one with mainecoon.

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/entity/pathfinding/PathJobFindTree.java [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 import com.minecolonies.entity.ai.citizen.lumberjack.Tree;
 import net.minecraft.block.Block;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/entity/pathfinding/PathJobFindTree.java]: MINOR Remove this unused import 'net.minecraft.block.Block'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/entity/pathfinding/AbstractPathJob.java [hunk 2]
```diff
@@ -900,23 +913,32 @@ else if (block.getMaterial(this.world.getBlockState(pos)).isLiquid())
...
      */
-    protected boolean isWalkableSurface(@NotNull Block block, BlockPos pos)
+    protected SurfaceType isWalkableSurface(@NotNull IBlockState blockState)
     {
-        return block.getMaterial(this.world.getBlockState(pos)).isSolid()
-                 && !(block instanceof BlockFence)
-                 && !(block instanceof BlockFenceGate)
-                 && !(block instanceof BlockWall)
-                 && !(block instanceof BlockHutField);
+        Block block = blockState.getBlock();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/entity/pathfinding/AbstractPathJob.java]: MINOR Local variable 'block' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by Kostronor** [comment]
:+1:

**Comment by Kostronor** [comment]
@cltnschlosser can we review this fast?

**Comment by cltnschlosser** [comment]
:shipit:

**Pull request merged**